homeassistant/components/climate/homematic.py

```diff
@@ -76,28 +86,26 @@ def operation_list(self):
     @property
     def current_humidity(self):
         """Return the current humidity."""
-        if not self.available:
-            return None
-        return self._data.get('ACTUAL_HUMIDITY', None)
+        for node in HM_HUMI_MAP:
+            if node in self._data:
+                return self._data[node]
 
     @property
     def current_temperature(self):
         """Return the current temperature."""
-        if not self.available:
-            return None
-        return self._data.get('ACTUAL_TEMPERATURE', None)
+        for node in HM_TEMP_MAP:
+            if node in self._data:
+                return self._data[node]
```

Comment:
With the changes proposed in https://github.com/danielperna84/pyhomematic/pull/59 you could reduce this to `return self.data[self._hmdevice._set_temp]` and save a few CPU cycles.